packages/pigeon/lib/objc_generator.dart

```diff
@@ -62,48 +62,64 @@ String _className(String? prefix, String className) {
+      : 'void(^)(${objcType.ptr}, NSError *_Nullable)';
 }
 
-const Map<String, String> _objcTypeForDartTypeMap = <String, String>{
-  'bool': 'NSNumber',
-  'int': 'NSNumber',
-  'String': 'NSString',
-  'double': 'NSNumber',
-  'Uint8List': 'FlutterStandardTypedData',
-  'Int32List': 'FlutterStandardTypedData',
-  'Int64List': 'FlutterStandardTypedData',
-  'Float64List': 'FlutterStandardTypedData',
-  'List': 'NSArray',
-  'Map': 'NSDictionary',
+class _ObjcPtr {
+  const _ObjcPtr({required this.baseName, required this.hasAsterisk});
+  final String baseName;
+  final bool hasAsterisk;
+  String get ptr => '$baseName${hasAsterisk ? ' *' : ' '}';
+}
+
+const Map<String, _ObjcPtr> _objcTypeForDartTypeMap = <String, _ObjcPtr>{
+  'bool': _ObjcPtr(baseName: 'NSNumber', hasAsterisk: true),
+  'int': _ObjcPtr(baseName: 'NSNumber', hasAsterisk: true),
+  'String': _ObjcPtr(baseName: 'NSString', hasAsterisk: true),
+  'double': _ObjcPtr(baseName: 'NSNumber', hasAsterisk: true),
+  'Uint8List':
+      _ObjcPtr(baseName: 'FlutterStandardTypedData', hasAsterisk: true),
+  'Int32List':
+      _ObjcPtr(baseName: 'FlutterStandardTypedData', hasAsterisk: true),
+  'Int64List':
+      _ObjcPtr(baseName: 'FlutterStandardTypedData', hasAsterisk: true),
+  'Float64List':
+      _ObjcPtr(baseName: 'FlutterStandardTypedData', hasAsterisk: true),
+  'List': _ObjcPtr(baseName: 'NSArray', hasAsterisk: true),
+  'Map': _ObjcPtr(baseName: 'NSDictionary', hasAsterisk: true),
+  'Object': _ObjcPtr(baseName: 'id', hasAsterisk: false),
 };
 
 String _flattenTypeArguments(String? classPrefix, List<TypeDeclaration> args) {
   final String result = args
       .map<String>(
-          (TypeDeclaration e) => '${_objcTypeForDartType(classPrefix, e)} *')
+          (TypeDeclaration e) => _objcTypeForDartType(classPrefix, e).ptr)
       .join(', ');
   return result;
 }
 
 String? _objcTypePtrForPrimitiveDartType(String? classPrefix, NamedType field) {
   return _objcTypeForDartTypeMap.containsKey(field.type.baseName)
-      ? '${_objcTypeForDartType(classPrefix, field.type)} *'
+      ? _objcTypeForDartType(classPrefix, field.type).ptr
       : null;
 }
 
 /// Returns the objc type for a dart [type], prepending the [classPrefix] for
 /// generated classes.  For example:
 /// _objcTypeForDartType(null, 'int') => 'NSNumber'.
-String _objcTypeForDartType(String? classPrefix, TypeDeclaration field) {
+_ObjcPtr _objcTypeForDartType(String? classPrefix, TypeDeclaration field) {
   return _objcTypeForDartTypeMap.containsKey(field.baseName)
       ? field.typeArguments.isEmpty
           ? _objcTypeForDartTypeMap[field.baseName]!
-          : '${_objcTypeForDartTypeMap[field.baseName]}<${_flattenTypeArguments(classPrefix, field.typeArguments)}>'
-      : _className(classPrefix, field.baseName);
+          : _ObjcPtr(
+              baseName:
+                  '${_objcTypeForDartTypeMap[field.baseName]!.baseName}<${_flattenTypeArguments(classPrefix, field.typeArguments)}>',
```

Comment:
This will have weird spacing as well without a fix for the issue mentioned earlier.